### ISSUE TITLE
Cry for a browser extension (Chrome and Firefox) -- JS coders, please!

### ISSUE THREAD

**Issue by nilsbyte** [shaarli/Shaarli#825]
This tool is awesome... if it only would have a proper browser extension. The extensions I found only replace the bookmarklet.

Things, a browser extension for Shaarli should have in my view:

- Login/logout (of course)
- Add links like with the bookmarklet via the extension symbol
- Search for tags (combined tags) and full-text via the Chrome omnibar
- Search for tags (combined tags) and full text via the extension symbol (popup)
- Edit found links via the extension symbol

Is something like this possible in the current state of development? I saw there is an API in development. Can it handle the things mentioned?

**Comment by alexisju** [comment]
Long time ago, i was also a regular user of this kind of add-on (with the del.icio.us plugin for Firefox), before to moving to Shaarli. Indeed i think that an intregrated browser based tool to manage (and use) saved links could be more efficient that a web based tool (in particular if you just want to keep and manage your bookmarks for yourself).

**Comment by ArthurHoaro** [comment]
I'm not sure to understand your request here. Would Shaarli results in the browser bar be mixed with your actual history and Google suggestion or would you use a special keyword? Would it returns the actual URL or a link to your shaare?

Could you provide one or two use case examples?

Also note that Shaarli supports OpenSearch, meaning that you can make a quick search by adding your instance as a search engine, apply a keyword like `shaa` and search with `shaa test`.

**Comment by alexisju** [comment]
In the past, for delicious, they were a browser addon which was a kind of sync and manage tool. All bookmarks were very quickly usable as local bookmarks.

Look at : http://web.archive.org/web/20111006160517/https://addons.mozilla.org/en-US/firefox/addon/delicious-bookmarks/

**Comment by nilsbyte** [comment]
@ArthurHoaro the Omnibar functionality usualy works in the way that you type a trigger, for example ´shaa´ and then the search string. The results are displayed among the browser history hits and Google suggest, mostly at the first position. When selecting a result, the underlying URL is opened.

The Chrome extension 'Bookmarks Tagger' (https://chrome.google.com/webstore/detail/bookmarks-tagger/dpiecafonfminhngabegejbligdagjfc) works this way. The trigger is ´bt´ here. The results are marked with a specific icon to tell them apart from suggests.

**Comment by nilsbyte** [comment]
@alexisju Yes, I am aware of this addon, but it did not sync traditional Browser-Bookmarks as far as I can remember. It only allowed for direct access to your tag-tree and the submission of URLs.

Bookmark Sync would NOT be a needed feature for a Shaarli browser extension, because there is no hierarchy or structure, only tags. Something like the tag-tree would of course be nice (see: http://web.archive.org/web/20111017153017/https://static-cdn.addons.mozilla.net/img/uploads/previews/full/24/24424.png?modified=1218135152), but not a necessity.

**Comment by virtualtam** [comment]
Ideally, it should follow the Browser Extension format to ease supporting compatible browsers:
- https://browserext.github.io/browserext/
- https://developer.chrome.com/extensions
- https://developer.mozilla.org/en-US/Add-ons/WebExtensions

**Comment by nodiscc** [comment]
I found https://gitlab.com/calfater/shaarli-synchronizer ([Firefox](https://addons.mozilla.org/en-US/firefox/addon/shaarli-synchronizer/?src=search), Chrome) which apparently uses the API to synchronize Firefox and Shaarli bookmarks.

I have not tested it yet but using the API instead of just relying on the "Add link" dialog (like the [Shaarli addon](https://addons.mozilla.org/en-US/firefox/addon/shaarli/) does) looks promising, as it would allow live synchronization and more.

**Comment by sunjam** [comment]
Have you consider Floccus?  It is a very nice extension, which supports Firefox/Chrome/Vivaldi.  Syncs Bookmarks and Folders across any number of devices to Nextcloud/WebDAV/local file system.  [See this related request in Floccus for Shaarli support](https://github.com/marcelklehr/floccus/issues/365).

[Here is documentation for adding additional backends](https://github.com/marcelklehr/floccus/blob/develop/doc/Adapters.md)
